R/compliance.R

```diff
@@ -0,0 +1,98 @@
+post_query <- function(url, params, bearer_token = get_bearer(), verbose = TRUE) {
+  bearer_token <- check_bearer(bearer_token)
+    r <- httr::POST(url, httr::add_headers(Authorization = bearer_token), body = params, encode = "json")
+    status_code <- httr::status_code(r)
+    if (!status_code %in% c(200)) {
+      stop(paste("something went wrong. Status code:", httr::status_code(r)))
+    }
+    r
+}
+
+#' Create Compliance Job
+#'
+#' This function creates a new compliance job and upload the Tweet IDs or user IDs.
+#' 
+#' @param file a text file, each line contains a Tweet ID or user ID.
+#' @param type the type of the job, whether "tweets" or "users".
+#' @param bearer_token string, bearer token
+#' 
+#' @return the input `file` (invisibly)
+#' @export
+#' 
+#' @examples
+#' \dontrun{
+#' create_compliance_job("tweetids.txt", "tweets")
+#' }
+create_compliance_job <- function(file,
+                                  type,
+                                  bearer_token = get_bearer()){
+  r <- post_query(url = "https://api.twitter.com/2/compliance/jobs",
+                  bearer_token,
+                  params = list("type" = type))
+  rcontent <- jsonlite::fromJSON(httr::content(r, "text"))
+  cat("Job ID: ", rcontent$data$id, "\n")
+  r <- httr::PUT(rcontent$data$upload_url,
+                 body = httr::upload_file(file))
+  if(r$status_code == 200){
+    cat("Upload Completed")
+  } else {
+    cat("Status Code: ", r$status_code)
```

Comment:
Should this be error instead?

Reply:
Probably changing it to something like "Error occured. Status Code:" would be better?